app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetController.scala

```diff
@@ -228,39 +232,52 @@ class EditSetController @Inject() (
     }
   }
 
-  def submit(id: String, recordId: String): Action[AnyContent] = Action { implicit request: Request[AnyContent] =>
+  def submit(id: String, oci: String): Action[AnyContent] = Action { implicit request: Request[AnyContent] =>
     withUser { user =>
       val title: String = resolvedMessage(MessageKeys.title)
+      logger.info(s"The edit set id is $id for record id $oci")
       val editSetName = editSets.getEditSet().name
-      logger.info(s"The edit set id is $id for record id $recordId")
 
-      request.body.asFormUrlEncoded.get("action").headOption match {
-        case Some("save") =>
-          formToEither(performAdditionalValidation(editSetRecordForm.bindFromRequest())) match {
-            case Left(formWithErrors) =>
-              BadRequest(
-                editSetRecordEdit(
-                  user,
-                  editSetName,
-                  title,
-                  legalStatus.getLegalStatusReferenceData(),
-                  corporateBodies.all,
-                  formWithErrors
-                )
-              )
-            case Right(editSetRecord) =>
-              editSetRecords.saveEditSetRecord(editSetRecord)
-              Redirect(controllers.routes.EditSetController.save(id, editSetRecord.oci))
-          }
-
-        case Some("discard") =>
-          Redirect(controllers.routes.EditSetController.discard(id, recordId))
-
-        case Some("calculateDates") => calculateDates(user, title)
-
-        // TODO Below added to handle error flow which could be a redirect to an error page pending configuration
-        case _ =>
+      val action = for {
```

Comment:
This was refactored because I needed to get an instance of the `EditSetRecord` by OCI - handling the case where a record was not found would have made this method too unwieldy in my opinion.

Reply:
Good call.